theano/gof/toolbox.py

```diff
@@ -440,10 +440,21 @@ def on_change_input(self, fgraph, node, i, r, new_r, reason=None):
 
 class NoOutputFromInplace(Feature):
 
+    def __init__(self, first_output_idx=0, last_output_idx=None):
+        self.first_idx = first_output_idx
+        self.last_idx = last_output_idx
+
     def validate(self, fgraph):
         if not hasattr(fgraph, 'destroyers'):
             return True
-        for out in list(fgraph.outputs):
+
+        if self.last_idx is None:
+            outputs_to_validate = list(fgraph.outputs)[self.first_idx:]
```

Comment:
Always using the else condition is good. By default, None as the stop of a slice mean the full slice.